decode.go

```diff
@@ -888,6 +894,34 @@ func (d *Decoder) decodeTime(ctx context.Context, dst reflect.Value, src ast.Nod
 	return nil
 }
 
+func (d *Decoder) castToDuration(src ast.Node) (time.Duration, error) {
+	if src == nil {
+		return 0, nil
+	}
+	v := d.nodeToValue(src)
+	if t, ok := v.(time.Duration); ok {
+		return t, nil
+	}
+	s, ok := v.(string)
+	if !ok {
+		return 0, errTypeMismatch(reflect.TypeOf(time.Duration(0)), reflect.TypeOf(v))
+	}
+	t, err := time.ParseDuration(s)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to parse duration")
+	}
+	return t, nil
+}
+
+func (d *Decoder) decodeDuration(ctx context.Context, dst reflect.Value, src ast.Node) error {
+	t, err := d.castToDuration(src)
+	if err != nil {
+		return err
```

Comment:
```suggestion
		return errors.Wrapf(err, "failed to convert to duration")
```

Reply:
This is already wrapped (if necessary) inside `castToDuration`

Reply:
Yes! That's right. However, we still need to wrap the `err` in order to get the stack trace printed correctly.

Reply:
Hm, are you saying I should wrap it twice? I'm currently matching the existing implementation of `decodeTime`, which also doesn't wrap the error a second time.

Reply:
Yes, it's need to improve debuggability ( please try to output error with `%+v` )
Oh, decodeTime's case is wrong. It's need to wrap...

Reply:
I added the call to `Wrapf` on both `decodeDuration` and `decodeTime`. I also added a test case so you can see the resulting messages. It looks like in most cases the stack trace is getting lost anyway at line 1059 when the error is unwrapped before being returned.